testing/thor/index.js

```diff
@@ -0,0 +1,155 @@
+       console.log(private_key);
+       finalizer.sign(private_key);
+   })
+   var signed_tx = finalizer.build();
+   document.getElementById("signed_tx_result").textContent = signed_tx.to_hex();
+}
+
+window.document.getElementById("clear_tx_btn").onclick = function(self) {
+   inputs = [];
+   outputs = [];
+   show_inputs();
+   show_outputs();
+}
+
+window.document.getElementById("refresh_btn").onclick = function(self) {
+   refresh();
+}
+
+function refresh() {
+   axios.get("https://localhost:8443/api/v0/utxo",{})
+         .then( result => {
+            console.log("downloaded new data", result.data);
+            filter_utxos(result.data)
+            show_utxos();
+   });
+}
+
+function show_utxos() {
+    var utxo_block = document.getElementById("utxo_block");
+    while (utxo_block.hasChildNodes()) {
+       utxo_block.removeChild(utxo_block.lastChild);
+    }
+    utxos.forEach(function(utxo, i) {
+        var li = document.createElement("li");
+        var a = document.createElement("a")
+        var t = document.createTextNode(utxo.out_value + " "
+              + utxo.in_txid.substr(0,3) + "..."
+              + utxo.in_txid.substr(utxo.in_txid.length - 3)
+              + ":" + utxo.in_idx
+        );
+        li.appendChild(a);
+        a.appendChild(t);
+        a.setAttribute("class","clickable");
+        a.onclick = function() { 
+           inputs.push(utxo);
+           show_inputs();
+        };
+        utxo_block.appendChild(li);
+    });
+}
+
+function show_inputs() {
+    var input_block = document.getElementById("tx_input_block");
+    clear_el(input_block);
+    var total = 0;
+    inputs.forEach(function(input, i) {
+        var li = document.createElement("li");
+        var a = document.createElement("a")
+        var t = document.createTextNode(
+              [input.out_value
+              ,input.in_txid
+              ,input.in_idx
+              ].join(" "));
+        total += parseInt(input.out_value);
+        li.appendChild(a);
+        a.appendChild(t);
+        a.onclick = function() { alert('hi!') };
+        input_block.appendChild(li);
+    })
+    document.getElementById("tx_input_total").textContent=total;
+}
+
+function filter_utxos(new_utxos) {
+   utxos = new_utxos.filter(function(utxo) {
+     return (utxo.out_addr === user_address_hex);
+   })
+}
+
+function show_outputs() {
+    var output_block = document.getElementById("tx_output_block");
+    var total = 0;
+    clear_el(output_block);
+    outputs.forEach(function(output, i) {
+        console.log(output);
+        var li = document.createElement("li");
+        var a = document.createElement("a")
+        var t = document.createTextNode("" + output.value + " " + output.address);
+        total += parseInt(output.value);
+        li.appendChild(a);
+        a.appendChild(t);
+        a.setAttribute("class","clickable");
+        a.onclick = function() { alert('hi!') };
+        output_block.appendChild(li);
```

Comment:
A less tedious, more maintainable way to go about it is to have a markup template for a list item somewhere in the HTML document, e.g. in an invisible `<div/>`. When a new item is created, clone it into a document fragment and populate the fragment with values, then append the fragment as a child to the list.

Reply:
Here were are clearing multiple `<li>` so I expect that amount of code and complexities would be pretty much similar, but thanks, for the idea, I'll use that if representation will be more complex.